### ISSUE TITLE
Frontend tech stack references

### ISSUE THREAD

**Issue by dabreegster** [acteng/schema#8]
# Vue

https://codesandbox.io/s/imwdx

**Comment by Robinlovelace** [comment]
# Vite

https://github.com/vitejs/vite

Source: Randomly browsing Tauri docs https://tauri.app/v1/guides/getting-started/setup/vite/

**Comment by Robinlovelace** [comment]
N. stars, including svelte.js

![image](https://user-images.githubusercontent.com/1825120/188327914-0e1f51bd-088f-4c14-aef4-b14a87fd54c3.png)

**Comment by dabreegster** [comment]
There's an entire corner of the internet dedicated to debating the merits of different frontend frameworks. https://www.youtube.com/watch?v=FQPlEnKav48 is one approachable and humorous example. What I'm trying to do here is find example code/projects that do something interesting with Mapbox/libre and demonstrate what "real code" using each framework looks like

**Comment by Robinlovelace** [comment]
Aha, I see. Yes, the question is more which ones have good integration with web maps...

**Comment by Robinlovelace** [comment]
Seems maplibre directions uses vite (and svelte?) in there (I have almost no idea what I'm looking at!): https://github.com/maplibre/maplibre-gl-directions/blob/main/vite.lib.config.ts

**Comment by dabreegster** [comment]
> the question is more which ones have good integration with web maps...

Not necessarily. Lots of what JS frameworks do is help manage state and propagate changes. If you adjust a filter for types of scheme shown or tick a box for showing advanced controls, what parts of the app need to change? Lots will be on the map, of course, but the Map{box,libre} API already nicely encapsulates state and handles changes. So how the JS framework plays with a "heavy" component like this is kind of interesting.

> Seems maplibre directions uses vite (and svelte?) in there (I have almost no idea what I'm looking at!): 

vite is just a build tool. If you want to go poking around at all the dozens of libraries out there, feel free, but it's not necessarily the best use of time. I'm just using this issue to find specific example projects that have a map using different JS frameworks.

**Comment by Robinlovelace** [comment]
One more link on this (sorry, couldn't resist!): https://docs.maptiler.com/svelte/maplibre-gl-js/how-to-use-maplibre-gl-js/

**Comment by Robinlovelace** [comment]
Another promising option: https://mapstore.geosolutionsgroup.com/mapstore/#/

**Comment by Robinlovelace** [comment]
And on GitHub: https://github.com/geosolutions-it/MapStore2
